kiota_http/middleware/redirect_handler.py

```diff
@@ -87,3 +71,139 @@ async def send(
+            url = url.copy_with(host=request.url.host)
+
+        # Facilitate relative 'Location' headers, as allowed by RFC 7231.
+        # (e.g. '/path/to/resource' instead of 'http://domain.tld/path/to/resource')
+        if url.is_relative_url:
+            url = request.url.join(url)
+
+        # Attach previous fragment if needed (RFC 7231 7.1.2)
+        if request.url.fragment and not url.fragment:
+            url = url.copy_with(fragment=request.url.fragment)
+
+        return url
+
+    def _redirect_headers(
+        self, request: httpx.Request, url: httpx.URL, method: str
+    ) -> httpx.Headers:
+        """
+        Return the headers that should be used for the redirect request.
+        """
+        headers = httpx.Headers(request.headers)
+
+        if not self._same_origin(url, request.url):
+            if not self.is_https_redirect(request.url, url):
+                # Strip Authorization headers when responses are redirected
+                # away from the origin. (Except for direct HTTP to HTTPS redirects.)
+                headers.pop("Authorization", None)
+
+            # Update the Host header.
+            headers["Host"] = url.netloc.decode("ascii")
+
+        if method != request.method and method == "GET":
+            # If we've switch to a 'GET' request, then strip any headers which
+            # are only relevant to the request body.
+            headers.pop("Content-Length", None)
+            headers.pop("Transfer-Encoding", None)
+
+        # We should use the client cookie store to determine any cookie header,
+        # rather than whatever was on the original outgoing request.
+        headers.pop("Cookie", None)
+
+        return headers
+
+    def _redirect_stream(
+        self, request: httpx.Request, method: str
+    ) -> typing.Optional[typing.Union[httpx.SyncByteStream, httpx.AsyncByteStream]]:
+        """
+        Return the body that should be used for the redirect request.
+        """
+        if method != request.method and method == "GET":
+            return None
+
+        return request.stream
+
+    def _same_origin(self, url: httpx.URL, other: httpx.URL) -> bool:
+        """
+        Return 'True' if the given URLs share the same origin.
+        """
+        return (
+            url.scheme == other.scheme and url.host == other.host
+            and self.port_or_default(url) == self.port_or_default(other)
+        )
+
+    def port_or_default(self, url: httpx.URL) -> typing.Optional[int]:
+        if url.port is not None:
+            return url.port
+        return {"http": 80, "https": 443}.get(url.scheme)
+
+    def is_https_redirect(self, url: httpx.URL, location: httpx.URL) -> bool:
+        """
+        Return 'True' if 'location' is a HTTPS upgrade of 'url'
+        """
+        if url.host != location.host:
+            return False
+
+        return (
+            url.scheme == "http" and self.port_or_default(url) == 80 and location.scheme == "https"
```

Comment:
You can technically have http or https on different ports (8080, 8443, etc...) I don't think the port comparison here should play a role, protocol/scheme yes

Reply:
Makes sense. Updated in https://github.com/microsoft/kiota-http-python/pull/20/commits/c1942d8825a7d21587a55022d005d97973a761d1